### PR TITLE
APM-1645: Add custom logging of handled exception events 

### PR DESCRIPTION
**Pull request by npalaska**
This PR does the following:
    
- [ ] End user can call Auklet.exception(Exception e) inside the catch block of handled exception. 
- [ ] Upon calling this function we send an event for an exception where it is called.

This functionality is solely up to the end user to decide which handled exceptions he/she want to get notified for when they occur.

### CODE REVIEW

File: src/main/java/io/auklet/agent/AukletExceptionHandler.java [hunk 1]
```diff
@@ -72,16 +42,42 @@ protected static AukletExceptionHandler setup() {
...
     }
 
-    private void setStackTrace(List<Object> stackTrace, String exceptionMessage){
-        this.stackTrace = stackTrace;
-        Messages.map.put("stackTrace", stackTrace);
-        Messages.map.put("timestamp", System.currentTimeMillis());
-        Messages.map.put("excType", exceptionMessage);
+    protected static synchronized void sendEvent(Throwable thrown) {
+        List<Object> list = new ArrayList<>();
+        System.err.print("Exception in thread \"" + Thread.currentThread().getName() + "\" ");
```
Comment (bleib1dj) [inline src/main/java/io/auklet/agent/AukletExceptionHandler.java]: Do we need this print statement?

File: src/main/java/io/auklet/agent/AukletExceptionHandler.java [hunk 2]
```diff
@@ -72,16 +42,42 @@ protected static AukletExceptionHandler setup() {
...
+        thrown.printStackTrace(System.err);
+
+        System.out.println("Exception message from app  " + thrown.getMessage());
```
Comment (bleib1dj) [inline src/main/java/io/auklet/agent/AukletExceptionHandler.java]: Do we need this print statement?

File: src/main/java/io/auklet/agent/AukletExceptionHandler.java [hunk 3]
```diff
@@ -72,16 +42,42 @@ protected static AukletExceptionHandler setup() {
...
+        }
+        setStackTrace(list, thrown.getMessage());
+
+        try {
+            byte[] bytesToSend = Messages.createMessagePack();
+            MqttMessage message = new MqttMessage(bytesToSend);
+            message.setQos(2);
+            Auklet.client.publish("java/events/" + Device.getOrganization() + "/" +
+                    Device.getClient_Username(), message);
+            System.out.println("Message published");
```
Comment (bleib1dj) [inline src/main/java/io/auklet/agent/AukletExceptionHandler.java]: Do we need this print statement?

File: src/main/java/io/auklet/agent/AukletExceptionHandler.java [hunk 4]
```diff
@@ -72,16 +42,42 @@ protected static AukletExceptionHandler setup() {
...
+
+        } catch (MqttException | NullPointerException e) {
+            e.printStackTrace();
+            System.out.println(e.getMessage());
```
Comment (bleib1dj) [inline src/main/java/io/auklet/agent/AukletExceptionHandler.java]: Do we need this print statement?

File: src/main/java/io/auklet/agent/AukletExceptionHandler.java [hunk 5]
```diff
@@ -72,16 +42,42 @@ protected static AukletExceptionHandler setup() {
...
+        try {
+            byte[] bytesToSend = Messages.createMessagePack();
+            MqttMessage message = new MqttMessage(bytesToSend);
+            message.setQos(2);
+            Auklet.client.publish("java/events/" + Device.getOrganization() + "/" +
+                    Device.getClient_Username(), message);
+            System.out.println("Message published");
+
+        } catch (MqttException | NullPointerException e) {
+            e.printStackTrace();
```
Comment (bleib1dj) [inline src/main/java/io/auklet/agent/AukletExceptionHandler.java]: Do we need this call?

**Comment by npalaska** [comment]
@bleib1dj I can probably remove these print statements at the moment but these print statements essentially will change to logging statements once we implement logging.

**Pull request merged**